src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -136,6 +137,29 @@ public static Course parseCourse(String course) throws ParseException {
         return new Course(trimmedCourse);
     }
 
+    /**
+     * Parses a {@code String seniority} into a {@code Seniority}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code seniority} is invalid.
+     */
+    public static Seniority parseSeniority(String seniority) throws ParseException {
+        requireNonNull(seniority);
+        String trimmedSeniority = seniority.trim();
+
+        try {
+            int seniorityInt = Integer.parseInt(trimmedSeniority.substring(trimmedSeniority.length() - 1));
+
+            if (!Seniority.isValidSeniority(seniorityInt)) {
+                throw new ParseException(Seniority.MESSAGE_CONSTRAINTS);
+            }
+
+            return new Seniority(seniorityInt);
+        } catch (NumberFormatException e) {
+            throw new ParseException(Seniority.MESSAGE_CONSTRAINTS);
```

Comment:
Maybe we could write a test (in this PR, or next PR) to parse in an invalid number like `1.0` or `2.4` to induce this error so that this line is covered?